packages/@aws-cdk/core/lib/cfn-output.ts

```diff
@@ -115,6 +116,35 @@ export class CfnOutput extends CfnElement {
     this._exportName = exportName;
   }
 
+  /**
+   * Return the `Fn.importValue` expression to import this value into another stack
+   *
+   * The returned value should not be used in the same stack, but in a
+   * different one. It must be deployed to the same environment, as
+   * CloudFormation exports can only be imported in the same Region and
+   * account.
+   *
+   * The is no automatic registration of dependencies between stacks when using
+   * this mechanism, so you should make sure to deploy them in the right order
+   * yourself.
+   *
+   * You can use this mechanism to share values across Stacks in different
+   * Stages. If you intend to share the value to another Stack inside the same
+   * Stage, the automatic cross-stack referencing mechanism is more convenient.
+   */
+  public get importValue() {
+    // We made _exportName mutable so this will have to be lazy.
+    return Fn.importValue(Lazy.stringValue({
+      produce: () => {
+        if (!this._exportName) {
+          throw new Error(`Add an exportName to the CfnOutput at '${this.node.path}' in order to use 'output.importValue'`);
```

Comment:
Can't we auto-allocate an export name in this case (similar to the one we allocate for auto-refs)?

Reply:
We could, but I'm not 100% comfortable doing that yet. Doesn't seem too onerous to start by expecting people to provide a name.

Reply:
ok